internal/command/machine/suspend.go

```diff
@@ -0,0 +1,100 @@
+package machine
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/superfly/fly-go"
+	"github.com/superfly/flyctl/internal/command"
+	"github.com/superfly/flyctl/internal/flag"
+	"github.com/superfly/flyctl/internal/flapsutil"
+	mach "github.com/superfly/flyctl/internal/machine"
+	"github.com/superfly/flyctl/iostreams"
+)
+
+func newSuspend() *cobra.Command {
+	const (
+		short = "Suspend one or more Fly machines"
+		long  = short + "\n"
+
+		usage = "suspend [<id>...]"
+	)
+
+	cmd := command.New(usage, short, long, runMachineSuspend,
+		command.RequireSession,
+		command.LoadAppNameIfPresent,
+	)
+
+	cmd.Args = cobra.ArbitraryArgs
+
+	flag.Add(
+		cmd,
+		flag.App(),
+		flag.AppConfig(),
+		selectFlag,
+		flag.Duration{
+			Name:        "wait-timeout",
+			Shorthand:   "w",
+			Description: "Duration to wait for individual Machines to be suspended.",
+			Default:     0 * time.Second,
+		},
+	)
+
+	return cmd
+}
+
+func runMachineSuspend(ctx context.Context) (err error) {
+	var (
+		io          = iostreams.FromContext(ctx)
+		args        = flag.Args(ctx)
+		waitTimeout = flag.GetDuration(ctx, "wait-timeout")
+	)
+
+	machines, ctx, err := selectManyMachines(ctx, args)
+	if err != nil {
+		return err
+	}
+
+	machines, release, err := mach.AcquireLeases(ctx, machines)
+	if err != nil {
+		return err
+	}
+	defer release()
```

Comment:
I know it is counterintuitive but `release` func must be called even in case acquiring leases fail to do so.  Otherwise in case of a partial error acquiring a set of leases, the non-failed leases won't be released.